File: frontend/src/api/userSequence.ts
/**
 * 用户序列管理API
 */
import request from './index'

export interface UserSequence {
  id: number
  user_id: number
  level: string
  unit_price: number
  created_at: string
  updated_at: string
}

export interface UserSequenceCreate {
  level: string
  unit_price: number
}

export interface UserSequenceUpdate {
  level?: string
  unit_price?: number
}

export interface UserSequenceListResponse {
  total: number
  items: UserSequence[]
}

/**
 * 创建用户序列
 */
export function createUserSequence(data: UserSequenceCreate): Promise<UserSequence> {
  return request.post('/api/v1/user-sequences', data)
}

/**
 * 获取用户序列列表
 */
export function getUserSequences(): Promise<UserSequenceListResponse> {
  return request.get('/api/v1/user-sequences')
}

/**
 * 获取用户序列详情
 */
export function getUserSequence(sequenceId: number): Promise<UserSequence> {
  return request.get(`/api/v1/user-sequences/${sequenceId}`)
}

/**
 * 更新用户序列
 */
export function updateUserSequence(sequenceId: number, data: UserSequenceUpdate): Promise<UserSequence> {
  return request.put(`/api/v1/user-sequences/${sequenceId}`, data)
}

/**
 * 删除用户序列
 */
export function deleteUserSequence(sequenceId: number): Promise<void> {
  return request.delete(`/api/v1/user-sequences/${sequenceId}`)
}

File: backend/app/schemas/user_sequence.py
"""用户序列相关模式"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from decimal import Decimal


class UserSequenceBase(BaseModel):
    """用户序列基础模式"""
    level: str = Field(..., min_length=1, max_length=50, description="序列等级，如：初级开发、中级开发、高级开发")
    unit_price: Decimal = Field(..., gt=0, description="单价（元/人天），用于产值计算")


class UserSequenceCreate(UserSequenceBase):
    """创建用户序列请求"""
    pass


class UserSequenceUpdate(BaseModel):
    """更新用户序列请求"""
    level: Optional[str] = Field(None, min_length=1, max_length=50, description="序列等级")
    unit_price: Optional[Decimal] = Field(None, gt=0, description="单价（元/人天）")


class UserSequenceResponse(BaseModel):
    """用户序列响应"""
    id: int
    user_id: int
    level: str
    unit_price: Decimal
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class UserSequenceListResponse(BaseModel):
    """用户序列列表响应"""
    total: int
    items: list[UserSequenceResponse]
